source/guide.md

```diff
@@ -0,0 +1,441 @@
+to. For example, running `sass input.scss output.css` from your terminal
+would take a single Sass file, `input.scss`, and compile that file to
+`output.css`.
+
+You can also watch individual files or directories with the `--watch`
+flag. The watch flag tells Sass to watch your source files for changes,
+and re-compile CSS each time you save your Sass. If you wanted to watch
+(instead of manually build) your `input.scss` file, you'd just add the
+watch flag to your command, like so:
+
+```
+sass --watch input.scss output.css
+```
+
+You can watch and output to directories by using folder paths as your
+input and output, and separating them with a colon. In this example:
+
+{% renderFile 'source/code-snippets/_homepage-sass-watch.md' %}
+
+Sass would watch all files in the `app/sass` folder for changes, and
+compile CSS to the `public/stylesheets` folder.
+
+<div class="sl-c-callout sl-c-callout--fun-fact">
+ 
+### 💡 Fun fact:
+
+Sass has two syntaxes! The SCSS syntax (`.scss`) is used most commonly. It's
+a superset of CSS, which means all valid CSS is also valid SCSS. The
+indented syntax (`.sass`) is more unusual: it uses indentation rather than
+curly braces to nest statements, and newlines instead of semicolons to
+separate them. All our examples are available in both syntaxes.
+
+</div>
+
+---
+
+## Variables
+
+Think of variables as a way to store information that you want to reuse
+throughout your stylesheet. You can store things like colors, font stacks,
+or any CSS value you think you'll want to reuse. Sass uses the `$` symbol
+to make something a variable. Here's an example:
+
+{% assign code = "$font-stack: Helvetica, sans-serif;
+$primary-color: #333;
+
+body {
+font: 100% $font-stack;
+color: $primary-color;
+}
+
+===
+
+$font-stack: Helvetica, sans-serif
+$primary-color: #333
+
+body
+font: 100% $font-stack
+color: $primary-color" | codeExample: true, 'scss' -%}
+
+{% assign example_id = 1 %}
+{% assign ui_id = 1 %}
+{% render 'code_example',
+code: code,
+example_id: example_id,
+ui_id: ui_id %}
+
+When the Sass is processed, it takes the variables we define for the
+`$font-stack` and `$primary-color` and outputs normal CSS with our
+variable values placed in the CSS. This can be extremely powerful when
+working with brand colors and keeping them consistent throughout the site.
+
+---
+
+## Nesting
+
+When writing HTML you've probably noticed that it has a clear nested and visual hierarchy. CSS, on the other hand, doesn't.
+
+Sass will let you nest your CSS selectors in a way that follows the same
+visual hierarchy of your HTML. Be aware that overly nested rules will
+result in over-qualified CSS that could prove hard to maintain and is
+generally considered bad practice.
+
+With that in mind, here's an example of some typical styles for a site's
+navigation:
+
+{% capture example_id %}
+{{ example_id | plus: 1 }}
+{% endcapture %}
+{% capture ui_id %}
+{{ ui_id | plus: 3 }}
+{% endcapture %}
```

Comment:
I don't know all the minutia of liquid, but would it work a bit more compact like this?

```suggestion
{% assign example_id = example_id | plus: 1 %}
{% assign ui_id = ui_id | plus: 3 %}
```

Does ui_id always increase by three? Would it make sense to create a filter that hides the magic number, and is instead like `| incrementId` and `| incrementUiId`?